CMakeLists.txt

```diff
@@ -223,13 +223,34 @@ if("${device.libraries}" STRGREATER "")
         INSTALL_DEPENDENCY(${LIB_DEST} ${device.libraries_${i}.name} ${URL} ${BRANCH} ${device.libraries_${i}.type})
         list(APPEND CODAL_DEPS "${device.libraries_${i}.name}")
     endforeach()
+endif()
+
+# and also from codal.json
+if("${codal.libraries}" STRGREATER "")
+    message("Installing app dependencies...")
+    set(CODAL_LIBS ${codal.libraries})
+
+    foreach(i ${CODAL_LIBS})
+        SET(BRANCH "NONE")
+        SET(URL "${codal.libraries_${i}.url}")
+        if("${codal.libraries_${i}.branch}" STRGREATER "")
+            SET(BRANCH "${codal.libraries_${i}.branch}")
+        endif()
+        if("${codal.target.branches.${URL}}" STRGREATER "")
+            SET(BRANCH "${codal.target.branches.${URL}}")
+            MESSAGE (STATUS "Override branch: ${BRANCH}")
+        endif()
 
-    foreach(dep ${CODAL_DEPS})
-        message("${BoldGreen}Using library: ${dep}${ColourReset}")
-        add_subdirectory("${PROJECT_SOURCE_DIR}/${LIB_DEST}/${dep}")
+        INSTALL_DEPENDENCY(${LIB_DEST} ${codal.libraries_${i}.name} ${URL} ${BRANCH} ${codal.libraries_${i}.type})
```

Comment:
Is this for additional dependencies only? I imagine this wouldn't work if you wanted to override the target library from codal.json?

Reply:
yeah, additional; I don't think we need overriding, or anyway I don't